packages/@uppy/provider-views/src/ProviderView/ProviderView.jsx

```diff
@@ -296,33 +228,90 @@ export default class ProviderView extends View {
     }
   }
 
-  async listAllFiles (path, files = null) {
-    files = files || [] // eslint-disable-line no-param-reassign
-    const res = await this.provider.list(path)
-    res.items.forEach((item) => {
-      if (!item.isFolder) {
-        files.push(item)
-      } else {
-        this.addFolder(item)
-      }
-    })
-    const moreFiles = res.nextPagePath
-    if (moreFiles) {
-      return this.listAllFiles(moreFiles, files)
+  async recursivelyListAllFiles (path, files = []) {
+    for (;;) {
```

Comment:
Would be nice if we could simplify away from infinite loop. Can't we pass `res.nextPagePath` to `recursivelyListAllFiles` as an argument?

Reply:
not sure how to do that, but i can try to prettify the loop a bit

Reply:
ok let me know if it looks better now